File: kotlin_language/src/com/android/kotlin_run_java.java
package com.android;

public class kotlin_run_java {
    public static void is() {
        System.out.println("运行的是java代码");
    }

    public static void main(String[] args) {
        new JavaRunKotlin().run(); // 运行的是Kotlin代码
        JavaRunKotlin.Companion.kotlinRun(); // Kotlin的静态方法
    }
}

File: kotlin_language/src/com/android/kotlin_run_java.kt
package com.android

// 运行java代码有关键字需要使用``运行方法
fun main() {
    kotlin_run_java.`is`()
}

class JavaRunKotlin {
    fun run() {
        println("运行的是Kotlin代码")
    }

    companion object {
        fun kotlinRun() {
            println("Kotlin的静态方法")
        }
    }
}